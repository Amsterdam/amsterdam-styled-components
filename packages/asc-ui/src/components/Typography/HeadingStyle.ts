import styled, { css, styledComponents } from '@datapunt/asc-core'
import TypographyStyle from './TypographyStyle'

type Props = {} & styledComponents.StyledProps<any>

export const HeaderStyleCSS = css`
  margin-top: 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`

export default styled(TypographyStyle)<Props>`
  ${HeaderStyleCSS}
`
