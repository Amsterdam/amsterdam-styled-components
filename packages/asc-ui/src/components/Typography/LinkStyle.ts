import styled, { css } from '@datapunt/asc-core'
import TypographyStyle from './TypographyStyle'
import { color } from '../../utils'
import { focusStyleText } from '../../utils/themeUtils'

export const LinkStyleCSS = css`
  color: ${color('primary')};
  display: inline-block;
  ${focusStyleText()}

  &:hover {
    color: ${color('secondary')};
  }
`

export default styled(TypographyStyle)`
  ${LinkStyleCSS}
`
