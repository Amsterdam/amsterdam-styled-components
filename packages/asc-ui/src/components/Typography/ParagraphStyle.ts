import styled, { css } from '@datapunt/asc-core'
import TypographyStyle from './TypographyStyle'
import { breakpoint } from '../../utils'

export type Props = {
  hasLongText?: boolean
}

export const ParagraphStyleCSS = css<Props>`
  margin-top: 0;
  color: inherit;
  ${({ hasLongText }) =>
    hasLongText &&
    css`
      @media screen and ${breakpoint('min-width', 'tabletS')} {
        font-size: 18px;
        line-height: 25px;
      }
    `}
`

export default styled(TypographyStyle)<Props>`
  ${ParagraphStyleCSS}
`
