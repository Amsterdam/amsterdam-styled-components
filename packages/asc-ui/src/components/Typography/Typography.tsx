import React from 'react'
import TypographyStyle, {
  Props as TypographyStyleProps,
} from './TypographyStyle'

type Props = {
  $as?: string
} & TypographyStyleProps &
  React.HTMLAttributes<HTMLElement>

const Typography: React.FC<Props> = ({
  children,
  gutterBottom,
  paragraph,
  element,
  color,
  fontSize,
  $as,
  ...otherProps
}) => (
  <TypographyStyle
    {...{
      gutterBottom,
      paragraph,
      element,
      color,
      fontSize,
    }}
    as={$as}
    {...otherProps}
  >
    {children}
  </TypographyStyle>
)

export default Typography
