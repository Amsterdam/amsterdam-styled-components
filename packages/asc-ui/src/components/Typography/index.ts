export { default as TypographtStyle } from './TypographyStyle'
export { default, Props as TypographyProps } from './Typography'
