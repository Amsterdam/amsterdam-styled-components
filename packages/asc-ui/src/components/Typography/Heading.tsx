import React from 'react'
import { styledComponents } from '@datapunt/asc-core'
import HeadingStyle from './HeadingStyle'

const Heading: React.FC<styledComponents.StyledProps<any>> = ({
  children,
  ...otherProps
}) => <HeadingStyle {...otherProps}>{children}</HeadingStyle>

export default Heading
