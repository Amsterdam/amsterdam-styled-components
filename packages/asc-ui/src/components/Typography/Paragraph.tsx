import React from 'react'
import ParagraphStyle, { Props } from './ParagraphStyle'

export { Props }

const Paragraph: React.FC<Props> = ({ children }) => (
  <ParagraphStyle>{children}</ParagraphStyle>
)

export default Paragraph
