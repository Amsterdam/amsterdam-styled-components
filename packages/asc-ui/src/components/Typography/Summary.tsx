import React from 'react'
import SummaryStyle from './SummaryStyle'
import { Props as ParagraphProps } from './Paragraph'

const Summary: React.FC<ParagraphProps> = ({ children, ...otherProps }) => (
  <SummaryStyle {...otherProps}>{children}</SummaryStyle>
)

export default Summary
