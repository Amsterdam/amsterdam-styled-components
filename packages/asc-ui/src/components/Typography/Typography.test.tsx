import * as React from 'react'
import { renderWithTheme } from '../../utils/withTheme'
import Typography from './Typography'
import 'jest-styled-components'

describe('Typography', () => {
  it('should render the typography types', () => {
    const component = renderWithTheme(
      <>
        <Typography element="h1">Heading one</Typography>
        <Typography element="h2">Heading two</Typography>
        <Typography element="h3">Heading three</Typography>
        <Typography element="h4">Heading four</Typography>
        <Typography element="h5">Heading five</Typography>
        <Typography element="h6">Heading six</Typography>
        <Typography element="a" href="/">
          I am a link!
        </Typography>
        <Typography element="p">
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab dolor
          dolorem esse illum magni&nbsp;
          <Typography element="a" href="/">
            I am a link!
          </Typography>
          &nbsp; optio porro qui ratione vitae?
        </Typography>
      </>,
    )
    expect(component).toMatchSnapshot()
  })
})
