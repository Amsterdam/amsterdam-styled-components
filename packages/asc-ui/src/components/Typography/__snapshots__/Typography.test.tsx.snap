// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render the typography types 1`] = `
Array [
  .c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: 700;
  font-size: 24px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 30px;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media screen and (min-width:540.02px) {
  .c0 {
    font-size: 30px;
    line-height: 38px;
  }
}

<h1
    class="c0"
  >
    Heading one
  </h1>,
  .c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: 700;
  font-size: 20px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 28px;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media screen and (min-width:540.02px) {
  .c0 {
    font-size: 24px;
    line-height: 30px;
  }
}

<h2
    class="c0"
  >
    Heading two
  </h2>,
  .c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: 700;
  font-size: 20px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 28px;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<h3
    class="c0"
  >
    Heading three
  </h3>,
  .c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: 700;
  font-size: 18px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 25px;
  margin-bottom: 8px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<h4
    class="c0"
  >
    Heading four
  </h4>,
  .c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: 400;
  font-size: 18px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<h5
    class="c0"
  >
    Heading five
  </h5>,
  .c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: 400;
  font-size: 18px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<h6
    class="c0"
  >
    Heading six
  </h6>,
  .c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: inherit;
  font-size: 1rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.25;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<a
    class="c0"
    href="/"
  >
    I am a link!
  </a>,
  .c1 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: inherit;
  font-size: 1rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.25;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: 400;
  font-size: 16px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 22px;
  margin-bottom: 20px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media screen and (min-width:540.02px) {
  .c0 {
    margin-bottom: 24px;
  }
}

<p
    class="c0"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab dolor dolorem esse illum magni 
    <a
      class="c1"
      href="/"
    >
      I am a link!
    </a>
      optio porro qui ratione vitae?
  </p>,
]
`;
